pkg/provider/provider.go

```diff
@@ -1370,9 +1428,15 @@ type patchConverter struct {
 // If a difference is present at the given path and the path matches one of the patterns in the database of
 // force-new properties, the diff is amended to indicate that the resource needs to be replaced due to the change in
 // this property.
-func (pc *patchConverter) addPatchValueToDiff(path []interface{}, v, old interface{}, inArray bool) error {
+func (pc *patchConverter) addPatchValueToDiff(path []interface{}, v, old, input interface{}, inArray bool) error {
 	contract.Assert(v != nil || old != nil)
 
+	// If there is no old input, then the only possible diff here is a delete. All other diffs must be diffs between
+	// old and new properties that are populated by the server.
+	if input == nil && v != nil {
```

Comment:
Though this will handle the case where a server-populated property is added or changed between the old and the new state, it won't handle the case where a server-populated property is deleted. In order to handle that case we'd also need to provider the old input value. This is easy enough to do that we should probably just bite the bullet and do it.

Reply:
I went ahead and did this.